web/controllers/account_controller.ex

```diff
@@ -17,28 +20,40 @@ defmodule ResearchResource.AccountController do
           user_data
           |> Enum.to_list
           |> Enum.map(fn {key, val} -> {String.to_atom(key), val} end)
-          render conn, "index.html", user_details: user_details
+          render conn, "index.html", user_details: user_details, changeset: changeset
       true ->
         redirect(conn, to: consent_path(conn, :new))
     end
   end
 
   def update(conn, %{"account" => account_params}) do
-    conn.assigns.current_user
-    |> RedcapHelpers.user_to_record
-    |> Enum.filter(fn {key, _val} -> key != :email end)
-    |> Enum.into(%{})
-    |> Map.merge(account_params)
-    |> @redcap_api.save_record
-    |> case do
-      {:ok, %HTTPoison.Response{status_code: 200}} ->
+    old_user =
+      conn.assigns.current_user
+      |> RedcapHelpers.user_to_record
+      |> Enum.filter(fn {key, _val} -> key != :email end)
+      |> Enum.into(%{})
+
+    case Repo.transaction(save_update(conn, old_user, account_params)) do
+      {:ok, _response} ->
         conn
         |> put_flash(:info, "Profile Updated")
         |> redirect(to: account_path(conn, :index))
-      _ ->
+      {:error, _failed_operation, %{errors: [email: {message, _}]}, changeset} ->
+        conn
+        |> put_flash(:error, "Email #{message}")
+        |> redirect(to: account_path(conn, :index, changeset: changeset))
+      {:error, _failed_operation, _failed_value, changeset} ->
         conn
         |> put_flash(:error, "Something went wrong")
-        |> redirect(to: account_path(conn, :index))
+        |> redirect(to: account_path(conn, :index, changeset: changeset))
     end
   end
+
+  defp save_update(conn, user_details, new_user_details) do
+    changeset = User.email_changeset(conn.assigns.current_user, new_user_details)
+
+    Multi.new
+    |> Multi.update(:postgres, changeset)
+    |> Multi.run(:redcap, fn _changes_so_far -> @redcap_api.save_record(Map.merge(user_details, new_user_details)) end )
```

Comment:
Interesting, I think a good use case for `Ecto.Multi` 👍 (haven't seen this before tbh).
Maybe pop in a little comment explaining the use case of if one fails, then the save won't happen.
(Just so that everyone who sees this section won't have to look up `Ecto.Multi`)